plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -71,13 +71,15 @@ def verify
 
     def build_selectors
       all = containers + init_containers
-      all.map { |c| build_selector_for_container(c) }.compact
+      all.each_with_index.map { |c, i| build_selector_for_container(c, index: i) }.compact
     end
 
     private
 
-    def build_selector_for_container(container)
-      dockerfile = container[:"samson/dockerfile"] || 'Dockerfile'
+    def build_selector_for_container(container, index:)
```

Comment:
`index` is a little confusing as an argument at first glance. What do you thing about passing a boolean `first` or `first_container` instead?